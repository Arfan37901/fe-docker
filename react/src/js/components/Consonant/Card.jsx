import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Tooltip from './Tooltip';

const CARD_STYLE = {
    WIDE: '3:2',
    SQUARE: '1:1',
    FULL: 'full-card',
};

const Card = (props) => {
    const {
        id,
        title,
        label,
        description,
        ctaLink,
        ctaLabel,
        image,
        bannerDescription,
        bannerFontColor,
        bannerBackgroundColor,
        bannerIcon,
        secondaryLabelText,
        allowBookmarking,
        isBookmarked,
        cardSavedIco,
        cardUnsavedIco,
        onClick,
        saveBookmarkText,
        unsaveBookmarkText,
        cardStyle,
        videoURL,
    } = props;

    const handleClick = (clickEvt) => {
        clickEvt.stopPropagation();
        onClick(id);
    };

    const setClassName = () => {
        const res = ['consonant-card'];
        const style = cardStyle.toLowerCase().trim();

        if (style === CARD_STYLE.FULL) res.push('consonant-card_full');
        if (style === CARD_STYLE.SQUARE) res.push('consonant-card_square');

        return res.join(' ');
    };

    return (
        <div className={setClassName()} id={id}>
            <div
                className="consonant-card--img"
                style={{ backgroundImage: `url("${image}")` }}>
                {
                    bannerDescription &&
                    bannerFontColor &&
                    bannerBackgroundColor &&
                    <span
                        className="consonant-card--banner"
                        style={({
                            backgroundColor: bannerBackgroundColor,
                            color: bannerFontColor,
                        })}>
                        {
                            bannerIcon &&
                            <img
                                src={bannerIcon}
                                alt=""
                                loading="lazy"
                                width="8"
                                className="consonant-card--banner-icon" />
                        }
                        {bannerDescription}
                    </span>
                }
                { videoURL && <span className="consonant-card--video-ico" />}
            </div>
            <div className="consonant-card--inner">
                {
                    label &&
                    <span className="consonant-card--label">{label}</span>
                }
                <h2
                    className="consonant-card--title"
                    dangerouslySetInnerHTML={{ __html: title }} />
                {
                    ![CARD_STYLE.SQUARE, CARD_STYLE.FULL]
                        .some(el => el === cardStyle.toLowerCase().trim()) &&
                        <Fragment>
                            <p
                                className="consonant-card--text"
                                dangerouslySetInnerHTML={{ __html: description }} />
                            <div className="consonant-card--footer-wrapper">
                                <div className="consonant-card--footer-info">
                                    {
                                        allowBookmarking &&
                                        <button
                                            data-tooltip-wrapper
                                            type="button"
                                            className={
                                                isBookmarked ?
                                                    'consonant-card--footer-btn consonant-card--footer-btn_active' :
                                                    'consonant-card--footer-btn'
                                            }
                                            onClick={handleClick}>
                                            {(cardSavedIco && cardUnsavedIco) ?
                                                <img
                                                    src={
                                                        isBookmarked ?
                                                            cardSavedIco :
                                                            cardUnsavedIco
                                                    }
                                                    width="16"
                                                    alt=""
                                                    loading="lazy" /> :
                                                <span className="consonant-card--bookmark-ico" />
                                            }
                                            <Tooltip text={
                                                isBookmarked ? unsaveBookmarkText : saveBookmarkText
                                            } />
                                        </button>

                                    }
                                    {
                                        secondaryLabelText &&
                                        <span className="consonant-card--secondary-text">{secondaryLabelText}</span>
                                    }
                                </div>
                                <a
                                    href={ctaLink}
                                    target="_blank"
                                    className="consonant-card--btn">{ctaLabel}
                                </a>
                            </div>
                        </Fragment>
                }
            </div>
        </div>
    );
};

export default Card;

Card.propTypes = {
    id: PropTypes.string.isRequired,
    ctaLink: PropTypes.string.isRequired,
    ctaLabel: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    label: PropTypes.string,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    bannerDescription: PropTypes.string,
    bannerFontColor: PropTypes.string,
    bannerBackgroundColor: PropTypes.string,
    bannerIcon: PropTypes.string,
    secondaryLabelText: PropTypes.string,
    allowBookmarking: PropTypes.bool.isRequired,
    isBookmarked: PropTypes.bool.isRequired,
    cardSavedIco: PropTypes.string.isRequired,
    cardUnsavedIco: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    saveBookmarkText: PropTypes.string,
    unsaveBookmarkText: PropTypes.string,
    cardStyle: PropTypes.string,
    videoURL: PropTypes.string,
};

Card.defaultProps = {
    bannerIcon: '',
    bannerDescription: '',
    bannerFontColor: '',
    bannerBackgroundColor: '',
    secondaryLabelText: '',
    label: '',
    saveBookmarkText: 'Save card',
    unsaveBookmarkText: 'Unsave card',
    cardStyle: '',
    videoURL: '',
};
