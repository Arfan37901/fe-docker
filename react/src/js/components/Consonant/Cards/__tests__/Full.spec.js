import { screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Card from '../Full';

import { DEFAULT_PROPS_FULL } from '../../Helpers/Testing/Constants/Card';

import makeSetup from '../../Helpers/Testing/Utils/Settings';

const setup = makeSetup(Card, DEFAULT_PROPS_FULL);

describe('Consonant/Card', () => {
    test('should renders banner correctly', () => {
        const {
            props: {
                overlays: {
                    banner: {
                        description: bannerDescription,
                        fontColor: bannerFontColor,
                        backgroundColor: bannerBackgroundColor,
                        icon: bannerIcon,
                    },
                },
            },
        } = setup();

        const bannerElement = screen.getByTestId('consonant-card--banner');
        const bannerIconElement = screen.getByTestId('consonant-card--banner-icon');

        expect(bannerElement).toHaveStyle({
            color: bannerFontColor,
            backgroundColor: bannerBackgroundColor,
        });
        expect(bannerElement).toHaveTextContent(bannerDescription);
        expect(bannerIconElement).toHaveAttribute('src', bannerIcon);
    });
    test('should renders with badge', () => {
        setup({ badgeText: 'some badge text' });

        const badgeElement = screen.queryByText('some badge text');

        expect(badgeElement).not.toBeNull();
    });
    describe('Check snapshots', () => {
        test('should renders with bookmarking', () => {
            const { tree } = setup({
                isBookmarked: true,
                allowBookmarking: true,
                disableBookmarkIco: true,
            });

            expect(tree).toMatchSnapshot();
        });
    });
});
