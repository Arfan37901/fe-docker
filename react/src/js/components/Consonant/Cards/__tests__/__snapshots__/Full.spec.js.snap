// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Card Check snapshots should renders with bookmarking 1`] = `
<div
  className="consonant-full-card"
  data-testid="consonant-card"
  id="3"
>
  <div
    className="consonant-full-card--img"
    data-testid="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"\\")",
      }
    }
  >
    <span
      className="consonant-full-card--banner"
      data-testid="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <div
        className="consonant-full-card--banner-icon-wrapper"
      >
        <img
          alt=""
          data-testid="consonant-card--banner-icon"
          loading="lazy"
          src="some-banner-icon"
        />
      </div>
      <span>
        Some Banner Description
      </span>
    </span>
    <span
      className="consonant-full-card--badge"
    >
      Some Badge Text
    </span>
    <a
      className="consonant-full-card--video-ico"
      href="some-video-videoURL"
      rel="noreferrer"
      tabIndex="0"
      target="_blank"
    />
  </div>
  <a
    className="consonant-full-card--inner"
    href="ctaLink"
    rel="noreferrer"
    tabIndex="0"
    target="_blank"
    title=""
  >
    <span
      className="consonant-full-card--label"
    >
      some details
    </span>
    <h2
      className="consonant-full-card--title"
    >
      3 - Infobit
    </h2>
  </a>
</div>
`;
